doc/source/zuul.rst

```diff
@@ -65,5 +66,354 @@ each commit is correctly tested.
+
+Those certificates must be maintained according to the security
+requirements and deployment specifics. In general it is preferred to use
+short living self-signed certificates for the Zookeeper cluster as well as
+LetsEncrypt certificates for Web access.
+
+User and account Management
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Generally Zuul does not support user accounts. It mainly communicates with
+git hosting systems with appropriate credentials and has no information about
+particular users proposing changes there.
+
+Zuul supports optional `Tenant Scoped REST API
+<https://zuul-ci.org/docs/zuul/discussion/tenant-scoped-rest-api.html>`_, but
+this is currently not enabled in the current installation.
+
+Operational accounts
+^^^^^^^^^^^^^^^^^^^^
+
+There are not granular operator accounts in Zuul installation. There is only one account allowing operate the system.
+
+Technical and M2M accounts
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Every component of Zuul only communicates to Zookeeper. For this Zookeeper
+client TLS certificate is used. No other technical or M2M accounts exist on
+the system.
+
+Communication Matrix (internal)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As mentioned above Zuul components communicate with each other only through
+Zookeeper. In order to keep communication simple those missing connections are
+not mentioned explicitly.
+
+.. list-table::
+
+   * - From \\ To
+     - zookeeper
+     - vault
+   * - nodepool-builder
+     - TLS(2281)
+     - TLS(8200)
+   * - nodepool-launcher
+     - TLS(2281)
+     - TLS(8200)
+   * - zuul-web
+     - TLS(2281)
+     - TLS(8200)
+   * - zuul-merger
+     - TLS(2281)
+     - TLS(8200)
+   * - zuul-executor
+     - TLS(2281)
+     - TLS(8200)
+   * - zuul-scheduler
+     - TLS(2281)
+     - TLS(8200)
+   * - zookeeper
+     - TLS(2888,3888)
+     - TLS(8200)
+
+Zookeeper protocol details can be found at `Zookeeper Internals <https://zookeeper.apache.org/doc/r3.6.0/zookeeperInternals.html>`_.
+
+Communication Matrix (external)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. list-table::
+
+   * - From \\ To
+     - SQL DB
+     - Git hosting
+     - Cloud
+   * - nodepool-builder
+     - N
+     - N
+     - [CLOUD_TLS]_
+   * - nodepool-launcher
+     - N
+     - N
+     - [CLOUD_TLS]_
+   * - zuul-web
+     - [DB_TLS]_
+     - [TLS]_
+     - N
+   * - zuul-merger
+     - N
+     - [SSH]_
+     - N
+   * - zuul-executor
+     - N
+     - [SSH]_
+     - [SSH]_
+   * - zuul-scheduler
+     - N
+     - N
+     - N
+
+.. [TLS] HTTPS encrypted (TLS) on port 443
+.. [SSH] SSH encrypted on custom port (depends on the git provider)
+.. [CLOUD_TLS] HTTPS encrypted (TLS) on port 443
+.. [DB_TLS] Database protocol, encrypted (TLS) (port depends on conrete DB type)
+
+Deployment Design
+=================
+
+Zuul is installed in an isolated Kubernetes cluster. As a mean of further
+security isolation SQL database and Zookeeper must be installaled dedicated
```

Comment:
this "must be installed" is a general statement...shouldn't we state how it is installed instead ?